src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue

```diff
@@ -30,14 +30,15 @@
             </router-link>
           </div>
 
-          <div v-if="!ui.isEditableLocked" @click="attemptSaveTree">
-            <router-link :to="editOrViewTreeJsUrl"
-                         class="btn btn-outline-secondary mr-2"
-                         :title="trans('flow-builder.click-to-toggle-editing')"
-            >
+          <router-link v-if="!ui.isEditableLocked"
+             :to="editOrViewTreeJsUrl"
+             event=""
+             :title="trans('flow-builder.click-to-toggle-editing')"
+             class="btn btn-outline-secondary mr-2"
+             :class="{active: isEditable}"
+             @click.native.prevent="handlePersistFlow(editOrViewTreeJsUrl)">
```

Comment:
Does this end up calling save even when toggling from View Mode into Edit Mode ?

Reply:
Good point. I think it does. Fixing this.